internal/kconfig/kconfig.go

```diff
@@ -0,0 +1,265 @@
+package kconfig
+
+import (
+	"bufio"
+	"compress/gzip"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/cilium/ebpf/btf"
+	"github.com/cilium/ebpf/internal"
+)
+
+// Find find a kconfig file on the host.
+// It first reads from /boot/config- of the current running kernel and tries
+// /proc/config.gz if nothing was found in /boot.
+// If none of the file provide a kconfig, it returns an error.
+func Find() (*bufio.Scanner, error) {
+	var s *bufio.Scanner
+
+	kernelRelease, err := internal.KernelRelease()
+	if err != nil {
+		return nil, fmt.Errorf("cannot get kernel release: %w", err)
+	}
+
+	path := fmt.Sprintf("/boot/config-%s", kernelRelease)
+	f, err := os.Open(path)
+	if err != nil {
+		f, err = os.Open("/proc/config.gz")
```

Comment:
This would be easier to do in Parse I think, by making it take `io.ReaderAt`. Then you can pass `io.NewLimitReader(ReaderAt, math.MaxUint64)`to the gzip reader.

Reply:
I pushed a commit, please take a look.